addons/sale_project/views/project_task_views.xml

```diff
@@ -81,12 +87,29 @@
                 <attribute name="options">{'always_reload': True}</attribute>
                 <attribute name="context">{'res_partner_search_mode': 'customer'}</attribute>
             </xpath>
-            <xpath expr="//field[@name='partner_phone']" position="after">
+            <xpath expr="//field[@name='partner_id']" position="after">
                 <field name="sale_order_id" attrs="{'invisible': True}" groups="sales_team.group_sale_salesman"/>
                 <field name="sale_line_id" groups="!sales_team.group_sale_salesman" string="Sales Order Item" options='{"no_open": True}' readonly="1" context="{'create': False, 'edit': False, 'delete': False}" attrs="{'invisible': ['|', ('allow_billable', '=', False), ('project_id', '=', False)]}"/>
                 <field name="sale_line_id" groups="sales_team.group_sale_salesman" string="Sales Order Item" options='{"no_create": True}' readonly="0" context="{'create': False, 'edit': False, 'delete': False, 'with_price_unit': True}" placeholder="Non-billable" attrs="{'invisible': ['|', ('allow_billable', '=', False), ('project_id', '=', False)]}"/>
                 <field name="allow_billable" invisible="1"/>
             </xpath>
+            <xpath expr="//field[@name='child_ids']/tree/field[@name='partner_id']" position="attributes">
+                <attribute name="invisible">context.get('should_not_display_partner')</attribute>
+            </xpath>
+            <xpath expr="//field[@name='depend_on_ids']/tree/field[@name='partner_id']" position="attributes">
+                <attribute name="invisible">context.get('should_not_display_partner')</attribute>
+            </xpath>
+        </field>
+    </record>
+
+    <record id="project_task_view_tree_main_base" model="ir.ui.view">
+        <field name="name">project.task.main.tree.inherit</field>
+        <field name="model">project.task</field>
+        <field name="inherit_id" ref="project.project_task_view_tree_main_base"/>
+        <field name="arch" type="xml">
+            <field name="partner_id" position="attributes">
+                <attribute name="invisible">context.get('should_not_display_partner')</attribute>
```

Comment:
same comment here

Reply:
`column_invisible` isn't working on main tree. it only works on tree inside form :pensive: